modules/freewheel-sspBidAdapter.js

```diff
@@ -0,0 +1,372 @@
+    if (config.format === 'intext-roll') {
+      config.iframeMode = 'dfp';
+    } else {
+      config.domId = 'freewheelssp_prebid_target';
+    }
+  }
+
+  var script = 'var config = {' +
+  '  preloadedVast:vast,' +
+  '  ASLoader:new window.com.stickyadstv.tools.ASLoader(' + bid.params.zoneId + ', \'' + getComponentId(bid.params.format) + '\')';
+
+  for (var key in config) {
+    // dont' send format parameter
+    // neither zone nor vastUrlParams value as Vast is already loaded
+    if (config.hasOwnProperty(key) && key !== 'format' && key !== 'zone' && key !== 'zoneId' && key !== 'vastUrlParams') {
+      script += ',' + key + ':"' + config[key] + '"';
+    }
+  }
+  script += '};' +
+
+  'window.com.stickyadstv.' + getAPIName(bid.params.format) + '.start(config);';
+
+  return script;
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: ['banner', 'video'],
+  aliases: ['stickyadstv'], //  former name for freewheel-ssp
+  /**
+  * Determines whether or not the given bid request is valid.
+  *
+  * @param {object} bid The bid to validate.
+  * @return boolean True if this is a valid bid, and false otherwise.
+  */
+  isBidRequestValid: function(bid) {
+    return !!(bid.params.zoneId);
+  },
+
+  /**
+  * Make a server request from the list of BidRequests.
+  *
+  * @param {BidRequest[]} bidRequests A non-empty list of bid requests which should be sent to the Server.
+  * @return ServerRequest Info describing the request to the server.
+  */
+  buildRequests: function(bidRequests, bidderRequest) {
+    // var currency = config.getConfig(currency);
+
+    var currentBidRequest = bidRequests[0];
+    if (bidRequests.length > 1) {
+      utils.logMessage('Prebid.JS - freewheel bid adapter: only one ad unit is required.');
+    }
+
+    var zone = currentBidRequest.params.zoneId;
+    var timeInMillis = new Date().getTime();
+    var keyCode = hashcode(zone + '' + timeInMillis);
+    var requestParams = {
+      reqType: 'AdsSetup',
+      protocolVersion: '2.0',
+      zoneId: zone,
+      componentId: getComponentId(currentBidRequest.params.format),
+      timestamp: timeInMillis,
+      pKey: keyCode
+    };
+
+    // Add GDPR flag and consent string
+    if (bidderRequest.gdprConsent) {
+      requestParams._fw_gdpr_consent = bidderRequest.gdprConsent.consentString;
+
+      if (typeof bidderRequest.gdprConsent.gdprApplies === 'boolean') {
+        requestParams._fw_gdpr = bidderRequest.gdprConsent.gdprApplies;
+      }
+    }
+
+    if (currentBidRequest.params.gdpr_consented_providers) {
+      requestParams._fw_gdpr_consented_providers = currentBidRequest.params.gdpr_consented_providers;
+    }
+
+    var vastParams = currentBidRequest.params.vastUrlParams;
+    if (typeof vastParams === 'object') {
+      for (var key in vastParams) {
+        if (vastParams.hasOwnProperty(key)) {
+          requestParams[key] = vastParams[key];
+        }
+      }
+    }
+
+    var location = utils.getTopWindowUrl();

```

Comment:
This function doesn't exist anymore in Prebid 3.x.  Can you take a look at the `bidderRequest.refererInfo` property (see more info [here](http://prebid.org/dev-docs/bidder-adaptor.html#referrers))?  It may have the information you were using here instead.

Reply:
updated.